src/backend/data/feed.js

```diff
@@ -40,6 +42,22 @@ class Feed {
     addFeed(this);
   }
 
+  /**
+   * Removes current Feed from the database.
+   * Returns a Promise
+   */
+  delete() {
+    removeFeed(this.id);
```

Comment:
Remove all posts that use this `this.id` for their `feed` too.  Also, you need to `return` the Promise here: `return removeFeed()`